python/dlisio/plumbing/errorhandler.py

```diff
@@ -0,0 +1,216 @@
+             of failure. Changing the behaviour of critical from raising an
+             Exception to logging would in this case mean that a partially
+             indexed file is returned by load.
+    major    Any issue that dlisio is able to recover from under heavy
+             developer's assumption. Developers haven't seen these errors in the
+             real world and don't know if recovery is ok. Usually would appear
+             when accessing attributes.
+
+             By default a warning is logged.
+    minor    Any issue that dlisio is able to recover from under light
+             developer's assumption. Developers have seen this error in real
+             files and generally believe that data is presented to the users
+             correctly.
+
+             May also appear when nearby data is spoiled, hence hinting user
+             to be cautious.
+
+             By default an info message is logged.
+    info     Issue doesn't contradict specification, but situation is peculiar.
+
+             By default a debug message is logged.
+
+    ======== ===================================================================
+
+    Error handler does not cover all the warnings and exceptions.
+    It is mostly intended to be used for data parsing, not for data
+    presentation. Thus it would be applicable, for example
+
+    * on data load and parsing in :py:func:`dlisio.load()`,
+      :py:func:`dlisio.logicalfile.load()` and
+      :py:func:`dlisio.plumbing.Frame.curves()`
+    * on object access :py:func:`dlisio.logicalfile.find()`,
+      :py:func:`dlisio.logicalfile.object()`
+    * on attribute access :py:attr:`dlisio.plumbing.Frame.description`
+
+    User still gets default behavior on other occasions, like warnings on
+    missing SUL or exceptions in :py:func:`dlisio.plumbing.Frame.curves()` in
+    case of invalid channel/frame relationships. Some of these might be get
+    implemented in the future.
+
+    Please also note that error handle doesn't redefine issues categories, it
+    only changes default behavior.
+
+    Attributes
+    ----------
+    info:     :py:class:`ErrorHandler.Actions`
+        Action for merely information message
+
+    minor:    :py:class:`ErrorHandler.Actions`
+        Action for minor specification violation
+
+    major:    :py:class:`ErrorHandler.Actions`
+        Action for major specification violation
+
+    critical: :py:class:`ErrorHandler.Actions`
+        Action for critical specification violation
+
+
+    Warnings
+    --------
+    Escaping errors is a good solution when user needs to read as much data as
+    possible, for example, to have a general overview over the file. However
+    user must be careful when using this mode during close inspection.
+    If user decides to accept errors, they must be aware that some returned data
+    will be spoiled. Most likely it will be data which is stored in the file
+    near the failure.
```

Comment:
Maybe this is too obvious to mention specifically?

Reply:
1. https://i.redd.it/nx8b897mg9g41.png
2. I wouldn't say that it's too obvious. If I try imaging myself as a dlisio user, there is no logical connection for me between concepts "I accepted errors" and "Something might be wrong in the data that I received"
In my mind it's "Error is reported at line 11, it means that I can trust lines 1-10", which doesn't necessary hold in case of, for example, last returned record of indirectly formatted data. If data itself is spoiled, we will never know.

Reply:
1) hahaha
2) fine :D